gpflow/utilities/ops.py

```diff
@@ -103,6 +103,23 @@ def square_distance(X, X2):
     return dist
 
 
+def difference_matrix(X, X2):
+    """
+    Returns (X - X2ᵀ)
+
+    This function can deal with leading dimensions in X and X2.
+    If X has shape [..., N, D] and X2 has shape [..., M, D], the
+    output will have shape [..., N, M, D].
```

Comment:
Note that this is different broadcasting behaviour from `square_distance`. @vdutor do you have any thoughts on what we might want to do here ?

Reply:
This should be fixed now.